src/background/component.js

```diff
@@ -0,0 +1,15 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+export class Component {
```

Comment:
Would you mind adding some https://jsdoc.app/ docstrings? 👉 👈 
Just a tldr; 
-> This class is responsible for X use it like Y 
-> Bonus points for adding type info to the functions.

Reply:
Not at all, but I have a followup where Some Things Change - do you care if I just add it there?